grc/core/generator/flow_graph.tmpl

```diff
@@ -378,6 +403,37 @@ def main(top_block_cls=$(class_name), options=None):
         #end for
     tb.Wait()
         #end if
+    #elif $generate_options == 'bokeh_gui'
+    serverProc = bokehgui.utils.create_server()
+    def killProc(signum, frame, tb):
+        tb.stop()
+        tb.wait()
+        serverProc.terminate()
+        serverProc.kill()
+    time.sleep(1)
+    try:
+        \# Define the document instance
+        doc = curdoc()
+        #if $flow_graph.get_option('author')
+        doc.title = "$title - $flow_graph.get_option('author')"
+        #else
+        doc.title = "$title"
+        #end if
+        session = push_session(doc, session_id="$flow_graph.get_option('id')",
+                               url = "http://localhost:5006/bokehgui")
+        \# Create Top Block instance
+        tb = top_block_cls(doc)
+        try:
+            tb.start()
+            signal.signal(signal.SIGTERM, functools.partial(killProc, tb=tb))
+            session.loop_until_closed()
+        finally:
+            print "Exiting the simulation. Stopping Bokeh Server"
+            tb.stop()
+            tb.wait()
+    finally:
+        serverProc.terminate()
+        serverProc.kill()
```

Comment:
So, I am not sure what exactly are you suggesting. Is it that instead of the new server, we can connect the new process in existing server?

As of now: Every time the server starts, it checks if the particular port is being used. If yes, it checks next one. It creates new server instance if any port in the sequence is free! And for each server instance, a unique Document is being served.